debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9ConstantPoolCommand.java

```diff
@@ -0,0 +1,527 @@
+					try {
+						if (i < ramCPCount) {
+							J9RAMConstantDynamicRefPointer ref = J9RAMConstantDynamicRefPointer.cast(ramCPP);
+							if (ref.value().notNull()) { // resolved
+								ramInfo = "!j9object " + ref.value().getHexAddress() + " (resolved)";
+							} else {
+								J9ObjectPointer exception = ref.exception();
+								if (exception.notNull()) {
+									J9ClassPointer clazz = J9ObjectHelper.clazz(exception);
+									J9ClassPointer throwable = J9VMConstantPoolClassRefAt(vm, J9VmconstantpoolConstants.J9VMCONSTANTPOOL_JAVALANGTHROWABLE).value();
+									if (exception.getAddress() == vm.voidReflectClass().classObject().getAddress()) {
+										ramInfo = "NULL (resolved)";
+									} else if (J9ClassHelper.isSameOrSuperClassOf(throwable, clazz)) { // resolved exception
+										ramInfo = "!j9object " + exception.getHexAddress() + " (exception)";
+									} else { // resolving by a thread object
+										ramInfo = "!j9object " + exception.getHexAddress() + " (resolving Thread)";
+									}
+								} else {
+									ramInfo = "(unresolved)";
+								}
+							}
+						}
+
+						J9ROMConstantDynamicRefPointer romRef = J9ROMConstantDynamicRefPointer.cast(romCPP);
+						SelfRelativePointer callSiteData = SelfRelativePointer.cast(romClass.callSiteData());
+						long callSiteCount = romClass.callSiteCount().longValue();
+						U16Pointer bsmIndicies = U16Pointer.cast(callSiteData.addOffset(4 * callSiteCount));
+						U16Pointer bsmData = bsmIndicies.add(callSiteCount);
+						long bsmIndex = romRef.bsmIndexAndCpType().rightShift((int)J9NonbuilderConstants.J9DescriptionCpTypeShift).bitAnd(J9NonbuilderConstants.J9DescriptionCpBsmIndexMask).longValue();
+
+						long argCount;
+						for (long j = 0; j < bsmIndex; j++) {
+							argCount = bsmData.at(1).longValue();
+							bsmData = bsmData.add(argCount + 2);
+						}
+						argCount = bsmData.at(1).longValue();
+						long bsmMethodRefIndex = bsmData.at(0).longValue();
+						romInfo = "#" + bsmMethodRefIndex;
+
+						J9ROMNameAndSignaturePointer nameAndSig = romRef.nameAndSignature();
+						value = String.format("%s:%s #%d(",
+								J9UTF8Helper.stringValue(nameAndSig.name()),
+								J9UTF8Helper.stringValue(nameAndSig.signature()),
+								bsmMethodRefIndex);
+						for (long j = 1; j <= argCount; j++) {
+							value += "#" + bsmData.add(1 + j).at(0).longValue();
+							if (j < argCount) {
+								value += ", ";
+							}
+						}
+						value += ")";
+					} catch (NoClassDefFoundError | NoSuchFieldException e) {
+						// The type J9RAMConstantDynamicRefPointer and its fields may be unknown to the core file being examined.
+						throw new CorruptDataException(e);
+					}
+				} else {
+					throw new CorruptDataException("Incorrect or unsupported constant pool entry type");
+				}
+
+				CommandUtils.dbgPrint(out, String.format("%-8d%-50s%-50s%-20s%s%n", i, romInfo, ramInfo, cpTypeToString.get(cpType), value));
```

Comment:
Please fold this long line and use `getOrDefault()` to properly handle unsupported types:
```
	CommandUtils.dbgPrint(out, String.format("%-8d%-50s%-50s%-20s%s%n",
			i, romInfo, ramInfo,
			cpTypeToString.getOrDefault(cpType, "Incorrect or unsupported type"),
			value));
```